### PR TITLE
Fix reading catalog files via pyradiosky

### PR DESCRIPTION
**Pull request by bhazelton**
## Description

The current handling for reading in catalog files via pyradiosky can fail silently because of a missing `else` clause. It also does not handle our new standard `skyh5` file formats.

I also made the version requirements for pyuvdata & pyradiosky consistent throughout the repo. This was prompted by an error in the RTD build.

## Motivation and Context

This fixes a bug reported in Slack by Bharat.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Reference simulation update or replacement
- [ ] Documentation change (documentation changes only)
- [ ] Version change
- [ ] Build or continuous integration change

## Checklist:

For all pull requests:
- [x] I have read the [contribution guide](CONTRIBUTING.md).
- [x] My code follows the code style of this project.

Bug fix checklist:
- [x] My fix includes a new test that breaks as a result of the bug (if possible).
- [x] All new and existing tests pass.
- [ ] I have checked that I reproduce the reference simulations or if there are differences they are explained below (if appropriate). If there are changes that are correct, I will update the reference simulation files after this PR is merged.
- [ ] I have checked (e.g., using the benchmarking tools) that this change does not significantly increase typical runtimes. If it does, I have included a justification in the comments on this PR.
- [x] I have updated the [CHANGELOG](https://github.com/RadioAstronomySoftwareGroup/pyuvsim/blob/master/CHANGELOG.md).

### CODE REVIEW

File: pyuvsim/simsetup.py [hunk 1]
```diff
@@ -688,30 +689,35 @@ def initialize_catalog_from_params(obs_params, input_uv=None, return_recarray=Tr
...
-                    vo_params["dec_column"] = source_params["dec_column"]
+                required_params = ["table_name", "id_column", "flux_columns"]
+                warn_params = {"ra_column": "RAJ2000", "dec_column": "DEJ2000"}
+                for param in required_params:
+                    if param not in source_params:
+                        raise ValueError(f"No {param} specified for {catalog}.")
+                    vo_params[param] = source_params[param]
+                for param, default in warn_params.items():
+                    if param not in source_params:
+                        warnings.warn(f"No {param} specified for {catalog}, using default.")
```
Comment (mkolopanis) [inline pyuvsim/simsetup.py:700]: should we print out what the defualt is here too?

File: pyuvsim/simsetup.py [hunk 2]
```diff
@@ -688,30 +689,35 @@ def initialize_catalog_from_params(obs_params, input_uv=None, return_recarray=Tr
...
+        elif catalog.endswith('skyh5'):
+            sky.read_skyh5(catalog)
         elif catalog.endswith('hdf5'):
-            sky.read_healpix_hdf5(catalog)
+            # In principal, this could either be a skyh5 or an old-style healvis hdf5 file
+            # Try skyh5 first, if that doesn't work, use the old method
+            try:
+                sky.read_skyh5(catalog)
+            except ValueError:
+                sky.read_healpix_hdf5(catalog)
```
Comment (mkolopanis) [inline pyuvsim/simsetup.py:715]: Should this _also_ be wrapped in a try/except or are we just banking on pyradiosky to raise from here if it is not a healpix hdf5 file?

**Review by mkolopanis — COMMENTED** [review]
Super simple Here. I think I have a few questions but other than that looking good.

File: pyuvsim/simsetup.py (hunk 1 above)
Comment (bhazelton) [reply pyuvsim/simsetup.py:700]: sure, I can add that. It's a column name.

File: pyuvsim/simsetup.py (hunk 2 above)
Comment (bhazelton) [reply pyuvsim/simsetup.py:715]: pyradiosky checks if it's not right and raises a useful error.

Comment (mkolopanis) [reply pyuvsim/simsetup.py:717]: okay that sounds good enough for me then.

**Review by mkolopanis — APPROVED** [review]
Looks good to me. I don't know what why coverage is unhappy now though.

**Comment by bhazelton** [comment]
The coverage report makes no sense. It says that `pyuvsim/__init__.py` is a new file, which it definitely is not. I don't know what's going on with codecov today, it's not making sense across several repos.

**Pull request merged**